src/components/dialog/dialog.js

```diff
@@ -221,13 +221,14 @@ class Dialog extends Modal {
 
   get closeIcon() {
     if (this.props.showCloseIcon) {
-      return <Icon
-        className="carbon-dialog__close"
+      return (<Icon
```

Comment:
I agree I prefer how you've written it. I wonder if this is a configuration in eslint.